clients/src/test/java/org/apache/kafka/clients/producer/internals/SenderTest.java

```diff
@@ -3232,7 +3231,7 @@ public void testProducerBatchRetriesWhenPartitionLeaderChanges() throws Exceptio
             assertTrue(client.hasInFlightRequests());
             client.respond(produceResponse(tp0, -1, Errors.NOT_LEADER_OR_FOLLOWER, 0));
             sender.runOnce(); // receive produce response, batch scheduled for retry
-            assertTrue(!futureIsProduced.isDone(), "Produce request is yet not done.");
+            assertFalse(futureIsProduced.isDone(), "Produce request is yet not done.");
```

Comment:
Looks to me this error message is inverted too right? it's stating the desired state, not the failure: if the assert fails, it's because the produce request **is done** when it shouldn't (not because "is yet not done"). Maybe just "Produce request shouldn't complete yet"...

Reply:
Good catch, I updated the messages